### DISCUSSION TITLE
Mitsubishi i-Miev support (aka Peugeot iOn / Citroën C-Zero)

### DISCUSSION THREAD

**Discussion by njh** [meatpiHQ/wican-fw#172] [General]
I own an Peugeot iOn (which is basically the same as a Mitsubishi i-Miev) and my goal is to periodically get State of Charge and Millage information out of the car, so my [OpenEVSE](https://github.com/OpenEVSE) charger knows how much to charge it.

The CAN Bus identifiers for Mitsubishi i-Miev are reasonably well known. And I have previously managed to get the data out by sending commands to a ELM327 adaptor, but I was frustrated by Bluetooth being unreliable and the need to have a device to bridge between bluetooth and the charger.

Some of the CAN bus resources I have found on the web are:
* https://github.com/plaes/i-miev-obd2
* https://docs.openvehicles.com/en/latest/components/vehicle_mitsubishi/docs/
* https://github.com/openvehicles/Open-Vehicle-Monitoring-System-3/blob/master/vehicle/OVMS.V3/components/vehicle_mitsubishi/src/vehicle_mitsubishi.cpp
* http://can4eve.bitplan.com/index.php/OBDLink_LX
* https://github.com/BITPlan/can4eve/tree/master/obdii/src/main/java/com/bitplan/triplet

I have previously compiled some of the data I am interested in, into spreadsheet:
![Screenshot 2024-08-13 at 18 14 30](https://github.com/user-attachments/assets/5a13fd91-859b-4b92-8f38-9adefb62f000)

It looks like someone previously looked into this in #65 but got stuck due to the lack of filtering support at the time.

**Comment by njh** [comment]
Successfully getting metrics out of WiCAN and on to MQTT using the filters feature.
The values are being sent out on to the CAN bus continuously, so you don't need to use PID:
<img width="640" alt="Screenshot 2024-08-13 at 15 15 35" src="https://github.com/user-attachments/assets/c2773be2-98fe-46bc-8d37-8e67ea89cf89">

This is what the result looks like, on the MQTT receiver:
<img width="198" alt="Screenshot 2024-08-14 at 08 45 48" src="https://github.com/user-attachments/assets/fe5912f2-269d-49ca-8cd9-ddedb5cc50ba">

It is a shame that these don't go to different MQTT topics, rather than arriving as single Key-Value JSON values.

**Reply by njh** [reply]
Now realised that @raphael303 has already been through all of this and got a working JSON file for the Mitsubishi i-MiEV in a comment here:

https://github.com/meatpiHQ/wican-fw/issues/66#issuecomment-1751960247

Would be great to get the Mitsubishi i-MiEV added to the WiCAN library of cars to make it easier to re-use other people's work.

**Comment by fowi4hjte** [comment]
What is the build year of your car?
Can you turn on the climate control and heater while charging with DC/AC?
I search for some way to turn on the heater without owning the climate control remote. No one seem to have find out how to send the same commands to the car the optional remote sends to the ETACS.

**Reply by njh** [reply]
Sorry, I missed this. My car is a 2012 Peugeot iOn. Sadly it doesn't have one of those RF remote controls that the i-Miev has to turn on climate control / heater remotely. It is very frustrating that you can't even turn on the heater, while plugged in (because you can't turn on the ignition while plugged in).

The CAN bus also shuts down when the ignition is off and the car isn't charging - not sure if it is possible to wake the bus up somehow.

**Reply by fowi4hjte** [reply]
I have found a way to turn on the heater with the OEM Mitsubishi MUT-3 SE software (the PSA Diagbox software is sort of a clone for that).
I have to find out if the current turning on of the heater is CAN or K-Line based because MUT-3 SE software just use both at same time.
I think you can enable the ability to turn on the heater while charging by updating the software in your car. Later model years can turn on their heater while charging (AC and DC charging). And because the DC switch is enabled in the car when charging, the only thing that could hold you back from turning on the heater is software.

**Reply by mguaylam** [reply]
@fowi4hjte sorry to highjack the thread but was wondering if something was known about that remote on the Mitsubishi.
